Add Prometheus TLS support (#957)
uyuni-project/uyuni-docs: modules/administration/pages/tls-certs-monitoring.adoc

[[tls-certs-monitoring]]
= TLS Certificates for Monitoring

By default, {productname} does not provide any certificates for securing monitoring configuration.
For providing security, you can import custom certificates, self-signed or signed by a third party certificate authority (CA).

This section demonstrates how to generate client/server certificates for Prometheus and Node exporter minions self-signed with SUSE Manager CA.

The host name of the TLS keys and certificates must match the fully qualified host name of the machine you deploy them on.
You can set the host names in the ``X509v3 Subject Alternative Name`` section of the certificate.
You can also list multiple host names if your environment requires it.

== Generate client/server keys and certificates for monitoring

To generate TLS keys and certificates for use with Prometheus server and node exporters you can follow these instructions:

> .Procedure: Create TLS keys and certificates

Reviewer: ```suggestion
.Procedure: Creating TLS keys and certificates
```

Author: Moved to monitoring

. On the {productname} Server, change to the ``ssl-build`` directory:
+
----
cd /root/ssl-build

> ----

Reviewer 2: Better not here. This directory is used for the SUMA TLS cert. It is managed by rhn-ssl-tool and has a special structure. If we put other files there a customer could by accident choose the wrong name and we have a wild mix SUMA and Monitoring certs. This can result in tons of L3s.

Let's suggest a new directory like /root/monitoring-ssl-build

Author: Moved to using `rhn-ssl-tool`.

. Create private key:
+
----
openssl genrsa -out server.key 2048
----

> . Create Certificate Signing Request (CSR) using the private Key.

Reviewer: ```suggestion
. Create Certificate Signing Request (CSR) using the private key.
```

Author: File dropped.

  An example OpenSSL configuration can be found below.

>   Please ensure that the set-cname parameter is the fully qualified domain name of your SUSE Manager Server.

Reviewer: ```suggestion
  Ensure that the set-cname parameter is the fully qualified domain name (FQDN) of your SUSE Manager Server.
```

+

> ----

Reviewer: I think we should add backslashes at the end a line if the command with the parameters continues with the next line.

openssl req -sha256 -text -config rhn-minion-openssl.cnf
    -new -key server.key -out server.csr
----
. Create and sign certificate with CA:
+
----
openssl ca -extensions req_server_x509_extensions -outdir ./
    -config rhn-ca-openssl.cnf -in server.csr -batch
    -cert RHN-ORG-TRUSTED-SSL-CERT -keyfile RHN-ORG-PRIVATE-SSL-KEY
    -days 365 -md sha256 -out server.crt

> ----

Reviewer 2: Uhh, you suggest to sign it with the official CA? I think this is ok, but in this case I would drop this whole chapter and handle every monitored client like a proxy and use the existing tooling we have.

In this case you can also use /root/ssl-build as directory.
You also do not need to describe how the ssl config should look like.

Author: You are right. For some reason my first tests with this tool hadn't worked for me.
Now retested and everything works as a charm. I think that's the best approach.
I'll clean up.

Author: Added shorter section about certificates generation to monitoring page. Thanks.

=== Example for OpenSSL configuration file

The example of the OpenSSL configuration file ``rhn-minion-openssl.cnf`` used before can be found here.

> Ensure that the CN and alt_names parameters are the fully qualified domain names of your minion.

Reviewer: ```suggestion
Ensure that the CN and alt_names parameters are the FQDN of your Salt client.
```

Author: Moved to monitoring and fixed.

You can also list multiple host names if your environment requires it.
+
----
[ req ]
default_bits            = 2048
distinguished_name      = req_distinguished_name
prompt                  = no
x509_extensions         = req_server_x509_extensions
req_extensions          = v3_req

[ req_distinguished_name ]
C                       = DE
ST                      = Bayern
L                       = Nuernberg

> O                       = SUSE

Reviewer: ```suggestion
O                       = Example Org
```

Author: Dropped.

> OU                      = SUSE

Reviewer: ```suggestion
OU                      = Example Org
```

Author: Dropped.

CN                      = minion.tf.local

> emailAddress            = galaxy-noise@suse.de

Reviewer: ```suggestion
emailAddress            = galaxy-noise@example.org
```
